mediapipe/graphs/instantmotiontracking/calculators/gl_animation_overlay_calculator.cc

```diff
@@ -342,32 +344,45 @@ bool GlAnimationOverlayCalculator::LoadAnimationAndroid(
       float By = V3y - V1y;
       float Bz = V3z - V1z;
       // Calculate cross product
-      triangle_normals[idx] = Ay * Bz - Az * By;
-      triangle_normals[idx + 1] = Az * Bx - Ax * Bz;
-      triangle_normals[idx + 2] = Ax * By - Ay * Bx;
-    }
-
-    // Normalize each triangle surface normal
-    for(int i = 0; i < len; i+=3) {
-      float product = 0.0;
-      product = product + triangle_normals[i] * triangle_normals[i];
-      product = product + triangle_normals[i+1] * triangle_normals[i+1];
-      product = product + triangle_normals[i+2] * triangle_normals[i+2];
-      float magnitude = sqrt(product);
-      triangle_normals[i] /= magnitude;
-      triangle_normals[i+1] /= magnitude;
-      triangle_normals[i+2] /= magnitude;
+      float normal_x = Ay * Bz - Az * By;
+      float normal_y = Az * Bx - Ax * Bz;
+      float normal_z = Ax * By - Ay * Bx;
+      // Add connected normal to each associated vertex
+      // It is also necessary to increment each vertex denominator for averaging
+      vertex_normals_sum[v1 * 3] += normal_x;
```

Comment:
nit: let's add a quick for-loop here and make this a little cleaner.  If we really wanted to, we could do that for our v's as well (v[0], v[1], v[2]), but I think we can leave those as-is for now to make things clearer